core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorRelationMetadata.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public final class ConnectorRelationMetadata
```

Comment:
How is this different from `ConnectorTableMetadata`?

Reply:
I initially used `ConnectorTableMetadata`, but determined the class is suitable for tables, but not for any relation.
Also, `ConnectorTableMetadata` contains information which is suitable for engine-connector conversation (like table properties), but doesn't make sense to provide this from one connector to the other.

Reply:
We also have `TableColumnsMetadata` that might be relevant. What is a "relation" anyway? And what kind of information do we need to expose that is common to all of them?

Reply:
Thanks for the pointer. I didn't think about `TableColumnsMetadata`.
It doesn't fit the purpose here, however. It is designed for handling redirections during information_schema listings and thus has optional columns. Also, it would tie those two otherwise rather unrelated interfaces together. 

> What is a "relation" anyway? 

table, view, or materialized view.

> And what kind of information do we need to expose that is common to all of them?

Currently we need column metadata. We may add more information in the future, if justified needs arise.

Reply:
We have a number of parallel classes for exposing metadata for these "table-like" objects:

```
ConnectorTableMetadata
ConnectorTableSchema
TableColumnsMetadata
```

In particular, ConnectorTableSchema is supposed to just contain the logical information for table-like objects (anything needed to be able to do analysis). ConnectorTableMetadata contains other information, such as properties, comments, etc.

My concern is that we're adding yet-another parallel API, and it just doesn't feel right.

Reply:
Replacing with `ConnectorTableSchema` for now, this addresses your concern @martint and fits perfectly the needs here.
My concern will be that we will now couple two otherwise unrelated ConnectorMetadata APIs with shared class used for information transport. Evolving the model may require a breaking change, but that's fine for me (it's a very seldomly implemented API and we can provide proper deprecation period), and seems not at all a concern for you.

Thanks for the feedback!